src/components/screens/timebank-view-all-screen.tsx

```diff
@@ -0,0 +1,290 @@
+        timespan: Timespan.ALL_TIME
+      });
+    } catch (error) {
+      setError(`${error} ${strings.error.totalTimeFetch}`);
+    }
+
+    return {
+      ...person,
+      personTotalTime: totalTime[0]
+    };
+  };
+
+  /**
+   * Fetches the person data & person total time data
+   */
+  const fetchPersonsAndPersonsTotalTime = async () => {
+    setLoading(true);
+
+    try {
+      let fetchedPersons: Person[] = persons;
+
+      if (!fetchedPersons.length) {
+        fetchedPersons = await personsApi.listPersons({
+          active: true
+        });
+        setPersons(fetchedPersons);
+      }
+
+      const personsTimeTotals = await Promise.all(
+        persons
+          .map((person) => ({
+            person: person
+          }))
+          .map(populatePersonTotalTimeData)
+      );
+
+      setPersonsWithTotalTime(personsTimeTotals);
+      setDisplayedPersonsTotalTime(personsTimeTotals);
+    } catch (error) {
+      setError(`${error} ${strings.error.fetchFailedGeneral}`);
+    }
+
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    fetchPersonsAndPersonsTotalTime();
+  }, []);
+
+  /**
+   * Renders the search
+   */
+  const renderSearch = () => (
+    <Box sx={{ display: "flex", justifyContent: "center" }}>
+      <TextField
+        value={searchInput}
+        onChange={handleSearchInputChange}
+        placeholder={strings.timebank.searchPlaceholder}
+        variant="standard"
+        disabled={loading}
+        sx={{ width: "99%", padding: 1 }}
+        InputProps={{
+          startAdornment: (
+            <InputAdornment position="start">
+              <Search />
+            </InputAdornment>
+          )
+        }}
+      />
+    </Box>
+  );
+
+  /**
+   * Handle search input change
+   *
+   * @param event input change event
+   */
+  const handleSearchInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const newSearchInput = event.target.value;
+    setSearchInput(newSearchInput);
+
+    if (newSearchInput === "") {
+      setDisplayedPersonsTotalTime(personsWithTotalTime);
+      return;
+    }
+
+    const newDisplayedPersonsTotalTime = personsWithTotalTime.filter((personsWithTotalTime) =>
+      `${personsWithTotalTime.person.firstName} ${personsWithTotalTime.person.lastName}`
+        .toLowerCase()
+        .includes(newSearchInput.toLowerCase())
+    );
+
+    setDisplayedPersonsTotalTime(newDisplayedPersonsTotalTime);
+  };
+
+  /**
+   * Get balance color
+   *
+   * @param personTotalTime person total time object
+   * @returns color string
+   */
+  const getBalanceColor = (personTotalTime: PersonTotalTime) =>
+    personTotalTime.balance > 0 ? theme.palette.success.main : theme.palette.error.main;
+
+  return (
+    <>
+      <Card sx={{ marginBottom: 2 }}>{renderSearch()}</Card>
+      <Grid container spacing={2} textAlign={"center"} marginBottom={20}>
+        {loading ? (
+          <CircularProgress sx={{ margin: "auto", mt: "5%", mb: "5%" }} />
```

Comment:
As we know this is currently taking a very long time to load all the data why don't we also add a string along with the circular progress, something like, "Please wait this could take a few moments..."?

Reply:
Added a text saying "Please wait, this could take a moment..." on top of the circular progressbar when loading.